### PR TITLE
[Translation Support] Script to create resources object from Airtable translations DB

### PR DESCRIPTION
**Pull request by micahtyong**
# In this PR 
This PR formalizes [task 1](https://www.notion.so/calblueprint/Translation-Support-Design-Document-7703018a8c4e403dab8aaa49d2a322e7#0edcc4efcc6f4b2298ca27d072994e55) of our [roadmap for translation support](https://www.notion.so/calblueprint/Translation-Support-Design-Document-7703018a8c4e403dab8aaa49d2a322e7) by creating a script that pulls down translations from the Airtable and formats it according to [react-i18next](https://react.i18next.com/guides/quick-start) conventions. This script is run any time a production build is made in order to scrape in translations locally for offline-friendly support. 

## How to review
Upon running `npm run translation-sync`, the output should be the following: 
```
> meepanyar@0.1.0 translation-sync /Users/micahyong/Desktop/blueprint/meepanyar
> node ./src/lib/i18next/getResources.js

Successful pulling in translations
```

Then in `src/lib/i18next/resources.js`, we should see the following output: 
```
export default {
  my: {
    translation: {
      'Cancel': 'ဖျက်သိမ်းမည်',
      'Close': 'ပိတ်မည်',
      'Incident Date (Date of Occurence)': 'ဖြစ်ရပ်သည့်နေ့',
      'Preventative Maintenance': 'ကြိုတင်ထိန်းသိမ်းမှု',
      'Remaining': 'ပမာဏ',
      'Spent [Money]': 'သုံးစွဲငွေ',
      'Submit': 'တင်သွင်းမည်',
      ... <a lot more rows> 
  }
}
```

The results should also be the same as #61 and #57. 

## Relevant Links
- [Design document for translation support](https://www.notion.so/calblueprint/Translation-Support-Design-Document-7703018a8c4e403dab8aaa49d2a322e7) 

### Related PRs
- [Fetching and parsing translations](https://github.com/calblueprint/meepanyar-node/pull/10). This PR can be closed after this one is merged. 
- #57 
- #61 

## Next steps
- Propagating this hook throughout the entire app (essentially comes down to replacing any hardcoded strings with our new `useIntl` hook). 
- Support fill-in-the-blank translations as outlined [here](https://www.notion.so/calblueprint/Translation-Support-Design-Document-7703018a8c4e403dab8aaa49d2a322e7#6c221ab8002e4ddaad0c1ee82c21dad8)

CC: @julianrkung

### CODE REVIEW

File: src/lib/i18next/resources.js [hunk 1]
```diff
@@ -1,255 +1,265 @@
...
+      "History": "စာရင်းသမိုင်း",
+      "Go to settings to change this.": "၎င်းကိုပြောင်းလဲရန် Settings သို့သွားပါ။",
+      "This customer is deactivated.": "ဤမီးအသုံးပြုသူအားရပ်ဆိုင်းထားသည်။",
+      "Account Inactive": "အကောင့်ပိတ်ထား",
+      "Your account has been deactivated by an admin of your site. Please contact an admin of your site for more information.": "သင့်ဆိုက်အဒ်မင်မှသင်၏အကောင့်ကိုပိတ်ထားလိုက်ပြီ။ ပိုမိုသိရှိလိုပါကသင့်ဆိုက်၏အဒ်မင်ကိုဆက်သွယ်ပါ။",
+      "Get started": "စလိုက်ကြစို့",
+      "Hello": "မင်္ဂလာပါ",
+      "Email": "Email",
+      "Enter [x] (Your [x])": "သင့်-",
+      "Forgot [x]": "-မေ့နေပါသလား",
```
Comment (julianrkung) [inline src/lib/i18next/resources.js:108]: What's the plan to account for the `[x]` instances? For example, there is `Enter [x] (Your [x])` and `Enter [x] (to add)`, where they result in different translations

File: package.json [hunk 2]
```diff
@@ -64,8 +64,9 @@
     "test": "react-scripts test",
     "eject": "react-scripts eject",
     "build-sw": "node ./src/custom-sw/sw-env.js && node ./src/custom-sw/sw-build.js",
+    "translation-sync": "node ./src/lib/i18next/getResources.js",
     "clean-cra-sw": "rm -f build/precache-manifest.*.js && rm -f build/service-worker.js",
-    "build": "react-scripts build && npm run build-sw && npm run clean-cra-sw",
+    "build": "react-scripts build && npm run build-sw && npm run translation-sync && npm run clean-cra-sw",
```
Comment (julianrkung) [inline package.json]: Shouldn't the `npm run translation-sync` run before the `react-scripts build`? Or else the `build/` dir produced for production would use the `resources.js` from before the sync right?

You can test this by building once after pointing to the `Mee Panyar Translations` base ID (I changed the table name and column names to work with the script). Then serving the production build via `serve -s build`. If you go to the site and do `devtools --> source` and search for `resources.js` you should see that the translations haven't updated to the translations from the new base ID

The production builds with the stale file (before the sync occurs), and so `build` needs to run twice for the production build to get the new translations.

File: src/lib/i18next/getResources.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ */
+const getAndSetTranslations = async () => {
+    try {
+        const resultData = await getAllRecords('Translation'); 
+        const resources = { my: { translation: resultData } }; 
+        const jsonResponse = JSON.stringify(resources, null, 2);
+        fs.writeFileSync('./src/lib/i18next/resources.js', "export default " + jsonResponse, 'utf8');
+        console.log('Successful pulling in translations'); 
+    } catch (err) {
+        console.err('Error when creating translations: ', err);
```
Comment (julianrkung) [inline src/lib/i18next/getResources.js]: `console.error` instead of `console.err`

**Review by julianrkung — CHANGES_REQUESTED** [review]
Found a small bug I think!

File: package.json (hunk 2 above)
Comment (micahtyong) [reply package.json]: Thank you for this explanation! This definitely cleared up some lingering misconceptions I had about production build scripts.

File: src/lib/i18next/resources.js (hunk 1 above)
Comment (micahtyong) [reply src/lib/i18next/resources.js:108]: I've outlined a solution in the design document [here](https://www.notion.so/calblueprint/Translation-Support-Design-Document-7703018a8c4e403dab8aaa49d2a322e7#6c221ab8002e4ddaad0c1ee82c21dad8). **TLDR**: we'll add an optional argument, `x`, that will nicely handle this case for us! I'll push this out in a follow-up PR!

Comment (julianrkung) [reply src/lib/i18next/resources.js:108]: Great, thanks :) sorry should've looked at the design doc first

File: src/lib/i18next/getResources.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+ * Format JavaScript object with translations, storing in ./resources.js. 
+ * formatted according to the conventions in https://react.i18next.com/ 
+ */
+const getAndSetTranslations = async () => {
+    try {
+        const resultData = await getAllRecords('Translation'); 
+        const resources = { my: { translation: resultData } }; 
+        const jsonResponse = JSON.stringify(resources, null, 2);
+        fs.writeFileSync('./src/lib/i18next/resources.js', "export default " + jsonResponse, 'utf8');
+        console.log('Successful pulling in translations');
```
Comment (julianrkung) [inline src/lib/i18next/getResources.js]: NIT: Successfully pulled in translations (or something similarly past tense)? Done pulling!

**Review by julianrkung — APPROVED** [review]
BEAUTIFUL thank you great work :^)

**Pull request merged**